kong/types.go

```diff
@@ -146,6 +146,47 @@ type Healthcheck struct {
 	Threshold *float64            `json:"threshold,omitempty" yaml:"threshold,omitempty"`
 }
 
+// HealthDataAddress represents the health data address of a target
+// +k8s:deepcopy-gen=true
+type HealthDataAddress struct {
+	Port   *int    `json:"port,omitempty" yaml:"port,omitempty"`
+	IP     *string `json:"ip,omitempty" yaml:"ip,omitempty"`
+	Health *string `json:"health,omitempty" yaml:"data,omitempty"`
+	Weight *int    `json:"weight,omitempty" yaml:"weight,omitempty"`
+}
+
+// HealthDataWeight represents the health data weight of a target
+// +k8s:deepcopy-gen=true
+type HealthDataWeight struct {
+	Total       *int `json:"total,omitempty" yaml:"total,omitempty"`
+	Available   *int `json:"available,omitempty" yaml:"available,omitempty"`
+	Unavailable *int `json:"unavailable,omitempty" yaml:"unavailable,omitempty"`
+}
+
+// HealthData represents the health data of a target
+// +k8s:deepcopy-gen=true
+type HealthData struct {
+	Host       *string             `json:"host,omitempty" yaml:"host,omitempty"`
+	Port       *int                `json:"port,omitempty" yaml:"port,omitempty"`
+	NodeWeight *int                `json:"nodeWeight,omitempty" yaml:"nodeWeight,omitempty"`
+	Weight     *HealthDataWeight   `json:"weight,omitempty" yaml:"weight,omitempty"`
+	Addresses  []HealthDataAddress `json:"addresses,omitempty" yaml:"addresses,omitempty"`
```

Comment:
I think you'll want this type definition to refer to an array of pointers i.e `[]*HealthDataAddress`

Reply:
Thanks for the review, updated the structure according to comment.